cirq-core/cirq/sim/simulator_base.py

```diff
@@ -86,22 +88,35 @@ def __init__(
         noise: 'cirq.NOISE_MODEL_LIKE' = None,
         seed: 'cirq.RANDOM_STATE_OR_SEED_LIKE' = None,
         ignore_measurement_results: bool = False,
+        split_untangled_states: bool = False,
     ):
         """Initializes the simulator.
 
         Args:
-           dtype: The `numpy.dtype` used by the simulation.
-           noise: A noise model to apply while simulating.
-           seed: The random seed to use for this simulator.
-           ignore_measurement_results: If True, then the simulation
-               will treat measurement as dephasing instead of collapsing
-               process. This is only applicable to simulators that can
-               model dephasing.
+            dtype: The `numpy.dtype` used by the simulation.
+            noise: A noise model to apply while simulating.
+            seed: The random seed to use for this simulator.
+            ignore_measurement_results: If True, then the simulation
+                will treat measurement as dephasing instead of collapsing
+                process. This is only applicable to simulators that can
+                model dephasing.
+            split_untangled_states: Optimizes simulation by running unentangled
+                qubit sets independently and merging those states at the end.
         """
         self._dtype = dtype
         self._prng = value.parse_random_state(seed)
         self.noise = devices.NoiseModel.from_noise_model_like(noise)
         self._ignore_measurement_results = ignore_measurement_results
+        self._split_untangled_states = split_untangled_states
+
+    @abc.abstractmethod
+    def _create_act_on_arg(
+        self,
+        initial_state: Any,
+        qubits: Sequence['cirq.Qid'],
+        logs: Dict[str, Any],
+    ) -> TActOnArgs:
+        """Creates an args"""
```

Comment:
Unfinished docstring? Also, the class docstring needs updated to reflect the new API (subclasses now need to implement this instead of `_create_act_on_args`, IIUC)